### PR TITLE
adding HOME into envs when init containers

### PR DESCRIPTION
**Pull request by mitnk**
closing #680

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -16,9 +16,9 @@ fi
 # not a part of youki workspace. For the reasoning behind this,
 # please check the docs and readme
 
-cargo build --verbose $TGT $1
+cargo build $TGT $1 $2
 cd ./runtimetest
-cargo build --verbose $TGT $1
+cargo build $TGT $1 $2
```
Comment (mitnk) [inline build.sh:21]: I found the output of `--verbose` is too verbose in regular developing activity. I think we can change to use them as needed. e.g.
```
$ ./build.sh
$ ./build.sh --verbose
$ ./build.sh --release
$ ./build.sh --verbose --release
```

a bit dirty but works. we can make this script mature if need later.

File: crates/libcontainer/src/syscall/linux.rs [hunk 2]
```diff
@@ -205,6 +205,29 @@ impl Syscall for LinuxSyscall {
         Some(user)
     }
 
+    fn get_pwdir(&self, uid: uid_t) -> Option<String> {
```
Comment (utam0k) [inline crates/libcontainer/src/syscall/linux.rs:208]: How about using `nix::unixstd::Iser`? I think that It is better to use nix than to implement it ourselves.
https://docs.rs/nix/latest/nix/unistd/struct.User.html

File: docs/src/user/basic_usage.md [hunk 3]
```diff
@@ -56,6 +56,26 @@ Then to start the original/normal Docker daemon, you can run
 sudo systemctl start docker
 ```
 
+#### Let docker permanently know youki as a runtime
```
Comment (utam0k) [inline docs/src/user/basic_usage.md:59]: 👍

**Review by utam0k — COMMENTED** [review]
Can I ask you to add the unit test?

File: build.sh (hunk 1 above)
Comment (utam0k) [reply build.sh:21]: 👍

**Comment by mitnk** [comment]
> Can I ask you to add the unit test?

Sure, do I have an existing test to refer on?

File: crates/libcontainer/src/syscall/linux.rs (hunk 2 above)
Comment (mitnk) [reply crates/libcontainer/src/syscall/linux.rs:208]: Oh, I'll check on it

**Comment by utam0k** [comment]
> > Can I ask you to add the unit test?
> 
> Sure, do I have an existing test to refer on?

If you search the code with `mod tests`, you can see various unit tests. If it looks too difficult, I'll send out a PR to add it, and you can review it.

**Comment by mitnk** [comment]
The code is ready for another check.

**Review by Furisto — COMMENTED** [review]
Hey @mitnk, you need to sign your commits in order to pass CI.

**Comment by mitnk** [comment]
rebased for sign-off.

**Review by Furisto — APPROVED** [review]

**Comment by Furisto** [comment]
@mitnk Thanks!

**Pull request merged**